lib/src/main/java/com/auth0/android/lock/PasswordlessLock.java

```diff
@@ -190,19 +193,32 @@ public Builder(Auth0 account, LockCallback callback) {
          * Finishes the construction of the Lock.Options and generates a new Lock instance
          * with those Lock.Options.
          *
+         * @param activity a valid Activity context
          * @return a new Lock instance configured as in the Builder.
          */
-        public PasswordlessLock build() {
+        public PasswordlessLock build(@NonNull Activity activity) {
             if (options.getAccount() == null) {
-                Log.e(TAG, "You need to specify the com.auth0.Auth0 object with the Auth0 Account details.");
-                throw new IllegalStateException("Missing Auth0 account information.");
+                Log.w(TAG, "com.auth0.Auth0 account details not defined. Trying to create it from the String resources.");
+                try {
+                    options.setAccount(new Auth0(activity));
+                } catch (IllegalArgumentException e) {
+                    throw new IllegalStateException("Missing Auth0 account information.", e);
+                }
             }
             if (callback == null) {
                 Log.e(TAG, "You need to specify the callback object to receive the Authentication result.");
                 throw new IllegalStateException("Missing callback.");
             }
             Log.v(TAG, "PasswordlessLock instance created");
-            return new PasswordlessLock(options, callback);
+
+            if (options.getAccount().getTelemetry() != null) {
+                Log.v(TAG, String.format("Using Telemetry %s (%s) and Library %s", Constants.LIBRARY_NAME, com.auth0.android.lock.BuildConfig.VERSION_NAME, BuildConfig.VERSION_NAME));
+                options.getAccount().setTelemetry(new Telemetry(Constants.LIBRARY_NAME, com.auth0.android.lock.BuildConfig.VERSION_NAME, BuildConfig.VERSION_NAME));
```

Comment:
if telemetry is required, maybe this can be handled in Auth0 itself? on `getTelemetry()` check if it's null and then create and save the default